browser/ntp_background_images/ntp_background_pref.h

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_NTP_BACKGROUND_IMAGES_NTP_BACKGROUND_PREF_H_
+#define BRAVE_BROWSER_NTP_BACKGROUND_IMAGES_NTP_BACKGROUND_PREF_H_
+
+#include <string>
+
+#include "base/values.h"
+#include "third_party/abseil-cpp/absl/types/variant.h"
+#include "url/gurl.h"
+
+namespace user_prefs {
+class PrefRegistrySyncable;
+}  // namespace user_prefs
+
+class PrefService;
+
+// NTPBackgroundPref gives easy access to values for NTP background from prefs.
+class NTPBackgroundPref final {
+ public:
+  static constexpr char kDeprecatedPrefName[] =
+      "brave.new_tab_page.custom_background_enabled";
+  static constexpr char kPrefName[] = "brave.new_tab_page.background";
+
+  enum class Type {
+    kBrave,  // Images that we supply.
+    kCustomImage,
+    kSolidColor,
+    kGradientColor
+  };
+
+  explicit NTPBackgroundPref(PrefService* service);
+  NTPBackgroundPref(const NTPBackgroundPref& pref) = delete;
+  NTPBackgroundPref& operator=(const NTPBackgroundPref& pref) = delete;
+  ~NTPBackgroundPref();
+
+  static void RegisterPref(user_prefs::PrefRegistrySyncable* registry);
+
+  // Try to migrate the old pref for custom background into this new pref.
+  void MigrateOldPref();
+
+  // Types
+  Type GetType() const;
+  void SetType(Type type);
+  bool is_brave_type() const { return GetType() == Type::kBrave; }
+  bool is_custom_image_type() const { return GetType() == Type::kCustomImage; }
+  bool is_solid_color_type() const { return GetType() == Type::kSolidColor; }
+  bool is_gradient_color_type() const {
+    return GetType() == Type::kGradientColor;
+  }
+
+  // Returns true when we should pick one item of selected type every time NTP
+  // opens.
+  bool ShouldUseRandomItem() const;
+  void SetShouldUserRandomItem(bool random);
+
+  // Return an item to use as NTP background.
+  absl::variant<GURL, std::string> GetSelectedItem() const;
```

Comment:
these methods are not used and there's no tests for them currently.

Reply:
Yep. It's not used for now. Added a test for SelectedValue(). But left RandomeValue() as it is since it's not implemented yet(it's stub for future use - there's a spec for this).